src/System.Memory/src/System/ReadOnlySpan.cs

```diff
@@ -29,8 +29,6 @@ public ReadOnlySpan(T[] array)
         {
             if (array == null)
                 ThrowHelper.ThrowArgumentNullException(ExceptionArgument.array);
-            if (default(T) == null && array.GetType() != typeof(T[]))
```

Comment:
There are multiple constructor that take array. It should be removed in all of them. Note that this code is slow span specific - your tests should catch it if you run them on slow span.